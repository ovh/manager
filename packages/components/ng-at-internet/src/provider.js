import OvhAtInternet from '@ovh-ux/ovh-at-internet';

/**
 * @ngdoc service
 * @require ATInternet smarttag.js library
 * @name atInternet
 * @description
 * Angular service wrapping the AtInternet tracking Javascript library.
 *
 * More informations : http://developers.atinternet-solutions.com/javascript-fr/bien-commencer-fr-javascript-fr/initialisation-du-tracker-javascript/
 */
/**
 * @ngdoc service
 * @require ATInternet smarttag.js library
 * @name atInternetProvider
 * @description
 * Provider allowing configuration for atInternet service.
 */
export default class NgAtInternet extends OvhAtInternet {
  $get() {
    return this;
  }
}
