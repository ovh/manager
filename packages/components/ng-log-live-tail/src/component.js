import template from './template.html';
import controller from './controller';

export default {
  controller,
  template,
  transclude: {
    tile: '?tile',
    description: '?description',
  },
  bindings: {
    logApiUrl: '<',
    logKind: '<',
    logKeys: '<',
  },
};
