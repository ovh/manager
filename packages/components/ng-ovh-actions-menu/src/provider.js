/**
 * @ngdoc service
 * @name actionsMenu.actionsMenuProvider
 *
 * @description
 * actionsMenuProvider allows developper to configure the path of the translation file.
 *
 * @example
 * <pre>
 *     angular.module("myManagerApp").config(function (actionsMenuProvider) {
 *          // set a new path for translations of the module
 *          actionsMenuProvider.setTranslationPath("/the/new/translation/path");
 *      });
 * </pre>
 */
export default function() {
  const self = this;
  let translationPath =
    '../bower_components/ovh-angular-actions-menu/dist/ovh-angular-actions-menu';

  /*= ====================================
    =            CONFIGURATION            =
    ===================================== */

  /**
   *  @ngdoc method
   *  @name actionsMenu.actionsMenuProvider#setTranslationPath
   *  @methodOf actionsMenu.actionsMenuProvider
   *
   *  @description
   *  Allows you to change the default location of the translation file of the module.
   *
   *  @param {String} path The new path of the translation file.
   *
   *  @return {String} The new configured translation path.
   */
  self.setTranslationPath = function(path) {
    if (path) {
      translationPath = path;
    }

    return translationPath;
  };

  /* -----  End of CONFIGURATION  ------*/

  /**
   *  @ngdoc service
   *  @name actionsMenu.service:actionsMenu
   *
   *  @description
   *  This service enable you to load translation file.
   */
  self.$get = /* @ngInject */ function(
    $translate /* , $translatePartialLoader */,
  ) {
    return {
      /**
       *  @ngdoc method
       *  @name actionsMenu.service:actionsMenu#
       *  @methodOf actionsMenu.service:actionsMenu
       *
       *  @description
       *  Load the translation file from the configured path.
       *
       *  @return {Promise} When the translation file is loaded.
       */
      loadTranslations() {
        // $translatePartialLoader.addPart(translationPath);
        return $translate.refresh();
      },
    };
  };
}
