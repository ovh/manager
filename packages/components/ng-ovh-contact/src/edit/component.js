import controller from './controller';
import template from './template.html';

export default {
  require: {
    ovhContactCtrl: '^ovhContact',
  },
  template,
  controller,
  bindings: {
    options: '=?ovhContactChoiceOptions',
    customList: '=?ovhContactChoiceCustomList',
  },
};
