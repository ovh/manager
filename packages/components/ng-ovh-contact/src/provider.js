import angular from 'angular';
import { forEach, has, keys, set, snakeCase } from 'lodash-es';

export default function() {
  const self = this;

  self.$get = /* @ngInject */ function $get(
    $q,
    $translate,
    OvhContact,
    OvhApiMe,
    CONTACT_EMAIL_REGEX,
    iceberg,
  ) {
    /**
     *  @ngdoc service
     *  @name ovhContact.service:ovhContact
     *
     *  @requires $q
     *  @requires $translate
     *  @requires OvhContact
     *  @requires OvhApiMe
     *
     *  @description
     *  The `ovhContact` service is the actual core of ovhContact module.
     *  This service manage the contacts of the connected user.
     */

    const ovhContactService = {};
    let contacts = [];
    let schemas = null;

    /**
     *  @private
     *  Get /me ovh api schema. Call GET /me.json.
     */
    function getApiSchemas() {
      if (!schemas) {
        return OvhApiMe.v6()
          .schema()
          .$promise.then((apiSchemas) => {
            schemas = apiSchemas;
            return schemas;
          });
      }
      return $q.when(schemas);
    }

    /**
     *  @private
     *  Convert nic address inforamtions to set contact.address informations.
     */
    function convertNicAddressToContactAddress(nic) {
      // keys are the contact address fields names / values are nic fields names
      const addressMapping = {
        line1: 'address',
      };
      const contactAddress = {};

      forEach(schemas.models['contact.Address'].properties, (value, key) => {
        if (has(addressMapping, key) && has(nic, addressMapping[key])) {
          contactAddress[key] = nic[addressMapping[key]];
        } else if (has(nic, key)) {
          contactAddress[key] = nic[key];
        } else {
          contactAddress[key] = null;
        }
      });

      return contactAddress;
    }

    /**
     *  @private
     *  Convert a nic to contact.
     */
    function convertNicToContact(nic) {
      // keys are the contact fields names / values are nic fields names
      const fieldMapping = {
        organisationName: 'organisation',
        lastName: 'name',
        gender: 'sex',
      };
      const contact = {};

      // iterate into contact model to set nic infos
      angular.forEach(
        keys(schemas.models['contact.Contact'].properties),
        (key) => {
          if (key === 'address') {
            contact.address = convertNicAddressToContactAddress(nic);
          } else if (key === 'birthDay') {
            if (new RegExp(/^(\d{1,2})\/(\d{1,2})\/(\d{4})$/).test(nic[key])) {
              const splittedDate = nic[key].split('/');
              contact[key] = splittedDate.reverse().join('-');
            } else {
              contact[key] = null;
            }
          } else if (
            fieldMapping[key] &&
            (has(nic, fieldMapping[key]) ||
              has(nic, fieldMapping[key.toLowerCase()]))
          ) {
            contact[key] =
              nic[fieldMapping[key]] ||
              nic[fieldMapping[key.toLowerCase()]] ||
              null;
          } else if (has(nic, key) || has(nic, key.toLowerCase())) {
            contact[key] = nic[key] || nic[key.toLowerCase()] || null;
          } else {
            contact[key] = null;
          }
        },
      );

      return contact;
    }

    /* ----------  CONTACT LIST MANAGEMENT  ----------*/

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#addContact
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Add a contact into contacts list.
     *
     *  @param {Object|OvhContact} contactOptions An object with options
     *  or an instance of OvhContact.
     *
     *  @return {OvhContact} The added contact.
     */
    ovhContactService.addContact = function addContact(contactOptions) {
      const addedContact =
        contactOptions instanceof OvhContact
          ? contactOptions
          : new OvhContact(contactOptions);
      contacts.push(addedContact);
      return addedContact;
    };

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#getContactList
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Get the entire list of contacts. This use api v7 to get all contacts details.
     *
     *  @param {Boolean} refresh Flag telling to fully recreate the contact list.
     *
     *  @return {Promise} That return an Array of OvhContact.
     */
    ovhContactService.getContactList = function getContactList(refresh) {
      const selff = this;

      if (refresh) {
        contacts = [];
      }

      return iceberg('/me/contact')
        .query()
        .expand('CachedObjectList-Pages')
        .execute()
        .$promise.then(({ data: contactsList }) => {
          // filter contact that are not already added
          // this avoid loosing contact object reference
          // then add contact to contact list (at given index)

          const contactsToAdd = contactsList.filter(
            (contact) => !contacts.some(({ id }) => contact.id === id),
          );

          forEach(contactsToAdd, (contact) => {
            selff.addContact(contact);
          });

          return contacts;
        });
    };

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#getCreationRules
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Get the rules for creating a new ovh contact.
     *
     *  @return {Object} Representing the rules for creating an new contact.
     */
    ovhContactService.getCreationRules = function getCreationRules() {
      let models = null;

      function updateCreationRules(rules) {
        angular.forEach(rules, (value, key) => {
          if (angular.isDefined(models[value.fullType])) {
            if (models[value.fullType]) {
              if (angular.isArray(models[value.fullType].enum)) {
                const enumId = models[value.fullType].id
                  .replace(/Enum$/, '')
                  .toLowerCase();
                set(rules, `${key}.values`, []);
                angular.forEach(models[value.fullType].enum, (enumValue) => {
                  rules[key].values.push({
                    label: $translate.instant(
                      [
                        ['country', 'language'].indexOf(enumId) > -1
                          ? enumId
                          : snakeCase(`ovh_contact_edit_${enumId}`),
                        enumValue,
                      ].join('_'),
                    ),
                    value: enumValue,
                  });
                });
              } else if (angular.isObject(models[value.fullType].properties)) {
                angular.forEach(
                  updateCreationRules(models[value.fullType].properties),
                  (subValue, subKey) => {
                    // eslint-disable-next-line no-param-reassign
                    rules[`${key}.${subKey}`] = subValue;
                  },
                );
              }
            }
          }

          // TODO API hack, phone is mandatory
          if (key === 'phone') {
            set(value, 'canBeNull', 0);
          } else if (key === 'email') {
            set(value, 'regularExpression', CONTACT_EMAIL_REGEX);
          }
        });
        return rules;
      }

      return getApiSchemas().then(({ models: schemaModels }) => {
        models = schemaModels;
        return updateCreationRules(models['contact.Contact'].properties);
      });
    };

    /* ----------  CONNECTED USER  ----------*/

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#getConnectedUser
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Get the connected user informations. Call GET /me API.
     *
     *  @return {Object} Representing the connected user.
     */
    ovhContactService.getConnectedUser = function getConnectedUser() {
      return OvhApiMe.v6().get().$promise;
    };

    /**
     *  @ngdoc method
     *  @name ovhContact.service:ovhContact#convertConnectedUserToContact
     *  @methodOf ovhContact.service:ovhContact
     *
     *  @description
     *  Convert the connected user nic to an ovh contact.
     *
     *  @return {Promise} That returns an OvhContact instance.
     */
    ovhContactService.convertConnectedUserToContact = function convertConnectedUserToContact() {
      return this.getConnectedUser().then((nic) =>
        getApiSchemas().then(() => new OvhContact(convertNicToContact(nic))),
      );
    };

    return ovhContactService;
  };
}
