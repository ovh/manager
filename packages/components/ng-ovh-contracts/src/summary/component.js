import template from './template.html';

const component = {
  bindings: {
    contracts: '<items',
    inlineContracts: '<inlineItems',
    model: '=',
    name: '@?',
  },
  template,
};

export default component;
