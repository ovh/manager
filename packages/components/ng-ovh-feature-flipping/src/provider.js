import FeatureFlipping from './service';

export default class NgOvhFeatureFlippingProvider {
  constructor() {
    this.applicationName = null;
  }

  /**
   * Set the application name that will request the 2API feature availability.
   *
   * @param {string} applicationName The current application name
   */
  setApplicationName(applicationName) {
    this.applicationName = applicationName;
  }

  /* @ngInject */
  $get($q, $http) {
    return new FeatureFlipping($q, $http, this.applicationName);
  }
}
