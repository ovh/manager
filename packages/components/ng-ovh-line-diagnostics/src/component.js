import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    lineNumber: '@',
    serviceName: '@',
  },
  template,
  controller,
};
