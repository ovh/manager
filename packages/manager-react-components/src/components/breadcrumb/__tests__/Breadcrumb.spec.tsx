import { vitest } from 'vitest';
import { waitFor } from '@testing-library/react';
import { render } from '../../../utils/test.provider';
import { Breadcrumb } from '../Breadcrumb.component';

const setupSpecTest = async ({ hideRootLabel }: { hideRootLabel?: boolean }) =>
  render(
    <Breadcrumb
      rootLabel="vRack services"
      appName="vrack-services"
      hideRootLabel={hideRootLabel}
    />,
  );

describe('breadcrumb component', () => {
  beforeEach(() => {
    vitest.mock('../../../hooks/breadcrumb/useBreadcrumb', () => ({
      useBreadcrumb: vitest.fn(({ hideRootLabel }) => [
        { label: 'vRack services', href: '/', hideLabel: hideRootLabel },
        { label: 'vRack service', href: '/:id', hideLabel: false },
        {
          label: 'vRack service listing',
          href: '/:id/listing',
          hideLabel: false,
        },
      ]),
    }));
  });

  it('should test valid', () => {
    expect(true).toBe(true);
  });
  it('should render 3 breadcrumb items when hideRootLabel is false', async () => {
    expect(true).toBe(true);
    const { container, asFragment } = await setupSpecTest({
      hideRootLabel: false,
    });
    const items = container.querySelectorAll('li');
    expect(items.length).toBe(3);
    expect(items[0]).toBeVisible();
    expect(items[1]).toBeVisible();
    expect(items[2]).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should hide root label when hideRootLabel is true', async () => {
    const { container, asFragment } = await setupSpecTest({
      hideRootLabel: true,
    });
    const items = container.querySelectorAll('li');
    expect(items.length).toBe(2);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render 3 breadcrumb items when hideRootLabel is not provided', async () => {
    const { container, asFragment } = await setupSpecTest({});
    const items = container.querySelectorAll('li');
    expect(items.length).toBe(3);
    expect(asFragment()).toMatchSnapshot();
  });
});
