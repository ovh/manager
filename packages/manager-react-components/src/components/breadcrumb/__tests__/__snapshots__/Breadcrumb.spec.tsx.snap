// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`breadcrumb component > should hide root label when hideRootLabel is true 1`] = `
<DocumentFragment>
  <nav
    class=""
  >
    <ol
      class="_breadcrumb__list_1jn0o_2"
    >
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <a
          class="_link_udbet_2"
          href="/:id"
        >
          vRack service
        </a>
        <span
          class="_breadcrumb-separator_1arog_11"
          role="presentation"
        />
      </li>
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <span
          aria-current="page"
          href="/:id/listing"
        >
          vRack service listing
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`breadcrumb component > should render 3 breadcrumb items when hideRootLabel is false 1`] = `
<DocumentFragment>
  <nav
    class=""
  >
    <ol
      class="_breadcrumb__list_1jn0o_2"
    >
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <a
          class="_link_udbet_2"
          href="/"
        >
          vRack services
        </a>
        <span
          class="_breadcrumb-separator_1arog_11"
          role="presentation"
        />
      </li>
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <a
          class="_link_udbet_2"
          href="/:id"
        >
          vRack service
        </a>
        <span
          class="_breadcrumb-separator_1arog_11"
          role="presentation"
        />
      </li>
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <span
          aria-current="page"
          href="/:id/listing"
        >
          vRack service listing
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`breadcrumb component > should render 3 breadcrumb items when hideRootLabel is not provided 1`] = `
<DocumentFragment>
  <nav
    class=""
  >
    <ol
      class="_breadcrumb__list_1jn0o_2"
    >
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <a
          class="_link_udbet_2"
          href="/"
        >
          vRack services
        </a>
        <span
          class="_breadcrumb-separator_1arog_11"
          role="presentation"
        />
      </li>
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <a
          class="_link_udbet_2"
          href="/:id"
        >
          vRack service
        </a>
        <span
          class="_breadcrumb-separator_1arog_11"
          role="presentation"
        />
      </li>
      <li
        class="_breadcrumb-item_vrq54_2"
      >
        <span
          aria-current="page"
          href="/:id/listing"
        >
          vRack service listing
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
